spec/system/admin/reports_spec.rb

```diff
@@ -108,20 +108,72 @@
     end
   end
 
-  it "orders and distributors report" do
-    login_as_admin_and_visit admin_reports_path
-    click_link 'Orders And Distributors'
-    click_button 'Go'
+  context "for a complete, paid order" do
+    let!(:ready_to_ship_order) { create(:order_ready_to_ship) }
 
-    expect(page).to have_content 'ORDER DATE'
-  end
+    before do
+      login_as_admin_and_visit admin_reports_path
+    end
+
+    it "generates the orders and distributors report" do
+      click_link 'Orders And Distributors'
+      click_button 'Go'
 
-  it "payments reports" do
-    login_as_admin_and_visit admin_reports_path
-    click_link 'Payment Reports'
-    click_button 'Go'
+      rows = find("table.report__table").all("thead tr")
+      table_headers = rows.map { |r| r.all("th").map { |c| c.text.strip } }
+
+      expect(table_headers).to eq([
+                                    [I18n.t("report_header_order_date"),
+                                     I18n.t("report_header_order_id"),
+                                     I18n.t("report_header_customer_name"),
+                                     I18n.t("report_header_customer_email"),
+                                     I18n.t("report_header_customer_phone"),
+                                     I18n.t("report_header_customer_city"),
+                                     I18n.t("report_header_sku"),
+                                     I18n.t("report_header_item_name"),
+                                     I18n.t("report_header_variant"),
+                                     I18n.t("report_header_quantity"),
+                                     I18n.t("report_header_max_quantity"),
+                                     I18n.t("report_header_cost"),
+                                     I18n.t("report_header_shipping_cost"),
+                                     I18n.t("report_header_payment_method"),
+                                     I18n.t("report_header_distributor"),
+                                     I18n.t("report_header_distributor_address"),
+                                     I18n.t("report_header_distributor_city"),
+                                     I18n.t("report_header_distributor_postcode"),
+                                     I18n.t("report_header_shipping_method"),
+                                     I18n.t("report_header_shipping_instructions")]
+                                           .map(&:upcase)
+                                  ])
```

Comment:
In my opinion it's better to use the rendered strings like `"SKU"` in specs. It's easier to read and less code. It also means that it becomes more obvious if we made a mistake in a translation. And it's not like we change these strings all the time. We can do that in Transifex without changing en.yml and without affecting these specs.

Reply:
Right, thank you for pointing this out - makes total sense, and I think I get the full length of [one of your previous comments](https://github.com/openfoodfoundation/openfoodnetwork/pull/8319#discussion_r725804103) now.

This file is to be updated soon (WIP) to add more tests on this report. I'll commit a change to replace the I18n assertions by simple strings on a separate PR :+1: